src/Extensions/Logging/Internal/EventSource/ActivityEnhensementObserver.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using Microsoft.Extensions.Options;
+using Microsoft.Omex.Extensions.Abstractions.Activities;
+using Microsoft.Omex.Extensions.Abstractions.Activities.Processing;
+
+namespace Microsoft.Omex.Extensions.Logging
+{
+	internal class ActivityEnhensementObserver : IActivityStartObserver
+	{
+		private readonly IOptions<OmexLoggingOptions> m_options;
+
+		public ActivityEnhensementObserver(IOptions<OmexLoggingOptions> options)
+		{
+			m_options = options;
+		}
+
+		public void OnStart(Activity activity, object? payload)
+		{
+			if (m_options.Value.AddObsoleteCorrelationToActivity)
+			{
+#pragma warning disable CS0618
+				if (activity.GetObsoleteCorrelationId() != null)
```

Comment:
I'm not sure how this is supposed to work. So if there is a correlation id already set a new one on start, but what happens if it's not, then with the current code it will remain null.

Reply:
No that's a mistake, thank you, I'll add some test to cover this case

Reply:
Fixed and added unit tests  for observers thanks a lot